### DISCUSSION TITLE
There is an error for analysis in PADForSFS

### DISCUSSION THREAD

**Discussion by shangliang** [MadAnalysis/madanalysis5#216] [Q&A]
Hi,

When I use the analysis in PADForSFS, I got the following error.

MA5-ERROR: Recasting card: invalid analysis (not present in the PAD, PADForMA5tune and PADForSFS): cms_sus_16_048
MA5-ERROR: Invalid recasting card
MA5-ERROR:    job submission aborted.

I use the command in bash,
./bin/ma5 -R madanalysis5_hadron_card.dat

And the content in the file madanalysis5_hadron_card.dat is,
set main.recast = on
set main.recast.store_events = False
import tag_1_pythia8_events.hepmc.gz as signal
set signal.xsection = 0.123
set main.recast.card_path = recasting_card.dat
submit ANALYSIS_0

And the above file recasting_card.dat is,
#atlas_exot_2018_05  vsfs on sfs_card_atlas_exot_2018_05.ma5 # ATLAS - 13 TeV - di-jet resonance with a photon (76.8/fb)
#atlas_exot_2018_06  vsfs on sfs_card_atlas_exot_2018_05.ma5 # ATLAS - 13 TeV - multijet + met [inclusive] (139/fb)
#atlas_susy_2016_07  vsfs on sfs_card_atlas_susy_2016_07.ma5 # ATLAS - 13 TeV - multijet + met (36.1/fb)
#atlas_conf_2019_040 vsfs onsfs_card_atlas_susy_2016_07.ma5 # ATLAS - 13 TeV - multijet + met (139/fb)
#atlas_susy_2017_04_2body vsfs on sfs_card_atlas_susy_2017_04.ma5 # ATLAS - 13 TeV - displaced vertices with opposite charge leptons, 2-body decays (32.8/fb)
#atlas_susy_2017_04_3body vsfs on sfs_card_atlas_susy_2017_04.ma5 # ATLAS - 13 TeV - displaced vertices with opposite charge leptons, 3-body decays (32.8/fb)
#atlas_susy_2018_31  vsfs on sfs_card_atlas_susy_2018_31.ma5 # ATLAS - 13 TeV - multibottoms + MET (139/fb)
#cms_exo_19_010      vsfs on sfs_card_cms_exo_19_010.ma5 # CMS   - 13 TeV - Disappearing tracks (140/fb)
#cms_exo_16_022      vsfs on sfs_card_cms_exo_19_022.ma5 # CMS   - 13 TeV - Displaced leptons (2.6/fb)
cms_sus_16_048      vsfs on sfs_card_cms_sus_16_048.ma5 # CMS   - 13 TeV - Soft dilepton (35.9/fb)

I am wondering how to solve this problem.

Thanks.

**Comment by jackaraz** [comment]
Hi @shangliang, would you like to take some time and use words to describe your issue? There are humans at the end of this line that you are asking your questions to; we are not machines and certainly not obliged to answer a bunch of copied and pasted things.

**Reply by shangliang** [reply]
Hi, I have updated the issue. Thank you very much.

**Comment by shangliang** [comment]
Firstly, I run the command in terminal,
./bin/ma5 -R madanalysis5_hadron_card.dat
Then, I get the error I shown above. And the content in the madanalysis5_hadron_card.dat file and in the recasting_card.dat file, I have shown them above.
So I am wondering how to solve this problem. I have installed PAD and PADForSFS.

**Reply by shangliang** [reply]
The  version is v 1.10.10

**Comment by jackaraz** [comment]
Hi @shangliang, I can not reproduce your problem. Can you please run madanalysis in debug mode and copy the **relevant portions** here? Also, please use the [issues section](https://github.com/MadAnalysis/madanalysis5/issues/new/choose) for code-based problems, there we have infrastructure in place to guide you through to provide correct information so that we can help you faster.

**Comment by shangliang** [comment]
Dear Jack,
Thank you very much. And I have run the MA5 in debug mode as you suggested
for PADForSFS analysis, I get the following error,

MA5-DEBUG (detect_python.ExtractInfo() in detect_python.py::L68):
sys.executable:
        ***@***.***/bin/python3.11

MA5-ERROR: python compiler not found. Please install it before using
MadAnalysis 5

MA5:      - Python                   [FAILURE]

MA5-ERROR: This package is a mandatory package: MadAnalysis 5 can not run
without it.


In fact, I have installed python on my computer. And for analysis in PAD,
MA5 run successfully.


Best,

Liang

Jack Y. Araz ***@***.***> 于2023年9月11日周一 11:43写道：

**Reply by jackaraz** [reply]
This is a completely different issue than what you reported above. Your Python libraries are not linked to the namespace ``python`` properly. You can either modify ``bin/ma5`` file and direct it to see whatever Python version you like to use, or you can fix your ``.bashrc`` (or ``.bash_profile`` depending on which system you are using) to map the specific Python version that you are using to namespace ``python``. The choice is yours. However, note that this is not about madanalysis; it's your system settings that you set, which I can not help beyond this. Also, madanalysis has not yet been tested with Python 3.11, so you might face some issues there.

**Comment by shangliang** [comment]
Dear Jack,

Thank you for your help. I will try to solve it as you suggested.

Best regards,
Liang

Jack Y. Araz ***@***.***> 于2023年9月11日周一 12:31写道：
